clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -213,33 +238,50 @@ public void ensureActiveGroup() {
             needsJoinPrepare = false;
         }
 
+        if (heartbeatThread == null) {
+            heartbeatThread = new HeartbeatThread();
+            heartbeatThread.start();
+        }
+
         while (needRejoin()) {
             ensureCoordinatorReady();
 
             // ensure that there are no pending requests to the coordinator. This is important
             // in particular to avoid resending a pending JoinGroup request.
-            if (client.pendingRequestCount(this.coordinator) > 0) {
+            if (state == MemberState.LEAVING || client.pendingRequestCount(this.coordinator) > 0) {
```

Comment:
Not sure I understand why this condition was added? `client.awaitPendingRequests()` doesn't do anything unless it has pending requests. Does this accomplish something?